src/main/java/eu/fasten/vulnerabilityproducer/utils/mappers/PurlMapper.java

```diff
@@ -73,7 +73,7 @@ public void inferPurls(Vulnerability v) {
         findMatchedPurl(mavenMap, v, patchedDate, baseRepo);
         findMatchedPurl(pypiMap, v, patchedDate, baseRepo);
 
-        if (v.getPurls().size() == 0) {
+        if (v.getPurls().size() == 0 && v.getBaseCpes() != null) {
```

Comment:
Would you please elaborate a bit on `v.getBaseCpes() != null`? Why is it added in this PR?

Reply:
That's an extra check that I added before we try to infer the purl from the `baseCpe` that we store for the vulnerability. I noticed from the logs often vulnerabilities did not have any `baseCpe` stored. In that case, we can just skip it.